src/mod_block_nonfriends.erl

```diff
@@ -0,0 +1,110 @@
+%%%-------------------------------------------------------------------
+%%% File    : mod_block_nonfriends.erl
+%%% Author  : Devin Sills <dsills@skillz.com>
+%%% Created : 28 Mar 2022 by Devin Sills <dsills@skillz.com>
+%%%
+%%%----------------------------------------------------------------------
+
+-module(mod_block_nonfriends).
+-author('dsills@skillz.com').
+
+-behaviour(gen_mod).
+
+%% gen_mod callbacks.
+-export([start/2, stop/1, reload/3, depends/2, mod_options/1]).
+
+%% hook handlers
+-export([filter_packet/1]).
+
+-include("logger.hrl").
+-include_lib("xmpp/include/xmpp.hrl").
+
+-define(HOOK_PRIORITY, 0).
+
+%%%===================================================================
+%%% Callbacks and hooks
+%%%===================================================================
+start(Host, _Opts) ->
+  ejabberd_hooks:add(user_send_packet, Host, ?MODULE, filter_packet, ?HOOK_PRIORITY)
```

Comment:
nit I think we should change the name of the filter_packet function since that's a different hook and I was confused

Reply:
good point! Renamed it